src/components/swap/SwapModalFooter.tsx

```diff
@@ -34,6 +37,39 @@ interface AnalyticsEventProps {
   tokenOutAmountUsd: string | undefined
   lpFeePercent: Percent
   swapQuoteReceivedDate: Date | undefined
+  routes: RoutingDiagramEntry[]
+}
+
+const formatRoutesEventProperties = (routes: RoutingDiagramEntry[]) => {
```

Comment:
i think if you do the `forEach` as a single loop and skip storing the intermediate variables like `routesPercentages` this would be easier to read.

i'm not stuck on that and it's up to you, but i find this a bit hard to follow right now.

Reply:
nice catch youre totally right, only one foreach is needed. done.